codebase/superdarn/src.bin/tk/plot/map_plot.1.12/make_fov.c

```diff
@@ -103,3 +103,78 @@ struct PolygonData *make_fov(double tval,struct RadarNetwork *network,
 
     return ptr;
 }
+
+
+struct PolygonData *make_fov_data(struct GridData *gptr,struct RadarNetwork *network,
+                                  int chisham,int old_aacgm) {
+
+    double rho,lat,lon;
+    int i,j,rn,bm;
+    float pnt[2];
+    int yr,mo,dy,hr,mt;
+    double sc;
+    int frang=180;
+    int rsep=45;
+    struct PolygonData *ptr=NULL;
+    struct RadarSite *site=NULL;
+    int st_id;
+    int maxrange=75;
+
+    TimeEpochToYMDHMS(gptr->st_time,&yr,&mo,&dy,&hr,&mt,&sc);
+
+    ptr=PolygonMake(sizeof(float)*2,NULL);
```

Comment:
Do we need to check if this returns NULL?

Reply:
Probably, although some error checking is already performed in the subsequently called `PolygonAddPolygon` and `PolygonAdd` functions.  A more thorough error handling would require much more significant changes throughout the rest of `map_plot` (and likely the other plotting binaries as well).

Reply:
I never see the harm in always checking because you never know if someone did error handling before even if you're just doing it for your one function — just a good C programming practice. 

Up to you.